vnext/Desktop.UnitTests/ScriptStoreTests.cpp

```diff
@@ -0,0 +1,56 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include <BaseScriptStoreImpl.h>
+#include <CppUnitTest.h>
+
+// Windows API
+#include <Windows.h>
+
+// Standard Library
+#include <cstdio>
+#include <cstdlib>
+
+using namespace facebook::jsi;
+using namespace Microsoft::VisualStudio::CppUnitTestFramework;
+
+using std::make_shared;
+using std::make_unique;
+using std::unique_ptr;
+
+namespace Microsoft::JSI::Test {
+
+TEST_CLASS (ScriptStoreIntegrationTest) {
+  TEST_METHOD(V8_PersistPreparedScript) {
+    unique_ptr<PreparedScriptStore> preparedScriptStore = nullptr;
+
+    char tempPath[MAX_PATH];
+    if (GetTempPathA(MAX_PATH, tempPath)) {
+      preparedScriptStore = make_unique<facebook::react::BasePreparedScriptStoreImpl>(tempPath);
+    } else {
+      Assert::Fail(L"Could not get temporary folder");
+    }
+
+    constexpr size_t fileSize = 4 * 1024 * 1024;
+    constexpr size_t bufferSize = fileSize + 1; // +1 for trailing \0
+    char *buffer = static_cast<char *>(std::malloc(bufferSize * sizeof(char)));
```

Comment:
Why are we using a raw buffer instead of an raii type like vector? Note you can still manipulate a raw vector buffer.

Reply:
Because of the string formatting, but @hansenyy makes a good point that we don't need proper JS for this test. Any type of content should work.